pkg/kv/kv_client_lww_test.go

```diff
@@ -0,0 +1,355 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package kv_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/util/hlc"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/errors"
+	"github.com/stretchr/testify/require"
+)
+
+// unwrapConditionFailedError unwraps an error into a kvpb.ConditionFailedError.
+// It fails the test if there is no error or if the error cannot be unwrapped
+// into the expected type.
+func unwrapConditionFailedError(t *testing.T, err error) *kvpb.ConditionFailedError {
+	t.Helper()
+	require.Error(t, err, "expected an error")
+
+	var conditionFailedErr *kvpb.ConditionFailedError
+	require.True(t, errors.As(err, &conditionFailedErr),
+		"expected error to unwrap to *kvpb.ConditionFailedError, got: %v", err)
+
+	return conditionFailedErr
+}
+
+func TestDeleteWithOriginTimestamp(t *testing.T) {
```

Comment:
Not for this PR, but I'm concurrently adding some client-level datadriven tests that we might be able to use for these tests.

Reply:
That would be great! These tests feel very verbose for what they are actually checking and the assertions are much weaker than the assertions in the data driven tests.